crates/iota/src/client_ptb/ptb.rs

```diff
@@ -57,35 +57,70 @@ pub struct Summary {
     pub gas_cost: GasCostSummary,
 }
 
-impl PTB {
-    /// Parses and executes the PTB with the sender as the current active
-    /// address.
-    pub async fn execute(self, context: &mut WalletContext) -> Result<String, Error> {
-        let res = self.execute_to_styled_str(context).await?;
-        println!("{}", res.ansi());
-        Ok(res.to_string())
+pub enum PTBCommandResult {
+    Preview(PTBPreview),
+    Summary(Summary),
+    CommandResult(IotaClientCommandResult),
+    DevInspect(Box<DevInspectResults>),
+    Json(serde_json::Value),
+    Help { long: bool },
+}
+
+impl PTBCommandResult {
+    pub fn to_styled_str(self) -> StyledStr {
+        StyledStr::from(self.to_string())
     }
+}
 
+impl std::fmt::Display for PTBCommandResult {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        match self {
+            Self::Preview(ptb_preview) => ptb_preview.to_string(),
+            Self::CommandResult(res) => res.to_string(),
+            Self::Summary(summary) => Pretty(summary).to_string(),
+            Self::DevInspect(dev_inspect_results) => {
+                Pretty(dev_inspect_results.as_ref()).to_string()
+            }
+            Self::Json(value) => {
+                serde_json::to_string_pretty(&value).map_err(|_| std::fmt::Error)?
+            }
+            Self::Help { long } => {
+                let mut buf = Vec::new();
+                if *long {
+                    ptb_description()
+                        .write_long_help(&mut buf)
```

Comment:
Why not keep using render_help/render_long_help?

Reply:
Because the print directly to the console and this is a display impl

Reply:
I don't think they do, no. `write_help` and `write_long_help` are actually deprecated and they say to use `render_help` and `render_long_help` instead.

Reply:
ah, you are correct, sorry. I misremembered. I did it just because of the return type. Fixed it anyway.